lib/ballot.go

```diff
@@ -80,6 +81,25 @@ func (b Ballot) IsWellFormed(networkID []byte) (err error) {
 		return
 	}
 
+	var confirmed, proposerConfirmed time.Time
+	if confirmed, err = sebakcommon.ParseISO8601(b.B.Confirmed); err != nil {
+		return
+	}
+	if proposerConfirmed, err = sebakcommon.ParseISO8601(b.ProposerConfirmed()); err != nil {
+		return
+	}
+
+	timeStart := time.Now().Add(time.Duration(-1) * BallotConfirmedTimeAllowDuration)
```

Comment:
I understand you intended duration time as timeEnd - timeStart = BallotConfirmedTimeAllowDuration.

However, the timeStart and TimeEnd variable based on different time.
Hence, `timeEnd - timeStart` is not `BallotConfirmedTimeAllowDuration` but `BallotConfirmedTimeAllowDuration + delta`

You'd better make it based on the same source. like this
```
timeNow := time.Now()
timeStart := timeNow.Add(time.Duration(-1) * BallotConfirmedTimeAllowDuration)
timeEnd := timeNow.Add(BallotConfirmedTimeAllowDuration)
```